### PR TITLE
Create table `actions_events_addkey`

### PR DESCRIPTION
**Pull request by forgxyz**
This PR closes issue #20 by parsing the `actions_events` table for instances of `AddKey`.
  
  
Note: the updated commit assumes the upstream `actions_events` table has been updated to `txn_hash`. As such, this should be merged after that PR.

### CODE REVIEW

File: models/core/actions_events_addkey.sql [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+addkey_events as (
+
+  select
+
+    action_id,
+    txn_hash,
+    block_timestamp,
+    action_data:access_key:nonce::float as nonce,
+    action_data:public_key::string as public_key,
+    action_data:access_key:permission::object as permission,
```
Comment (chuxinh) [inline models/core/actions_events_addkey.sql:28]: i got an error when running this saying `failed to cast variant value `FullAccess` to OBJECT` which seems to apply to single string only?

File: models/core/actions_events_addkey.sql [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    action_data:access_key:permission:FunctionCall:allowance::float as allowance,
```
Comment (chuxinh) [inline models/core/actions_events_addkey.sql:29]:
```suggestion
    action_data:access_key:permission:FunctionCall:allowance::number as allowance,
```

File: models/core/actions_events_addkey.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
+version: 2
+
+models:
+  - name: actions_events_addkey
+    description: |-
+      This table extracts all action events from a transaction and stores the argument data under action_data.
```
Comment (chuxinh) [inline models/core/actions_events_addkey.yml:6]: update it to be related to `AddKey` actions only?

File: models/core/actions_events_addkey.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    columns:
+      - name: action_id
+        description: The `action_id` as compiled from `tx_id` and `action_index`. This is unique for each record.
+        tests:
+          - unique
+          - not_null
+
+      - name: txn_hash
+        description: The hash of the transaction, this is the primary key for this table.
```
Comment (chuxinh) [inline models/core/actions_events_addkey.yml:16]: `action_id` should be the primary key?

File: models/core/actions_events_addkey.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+      - name: nonce
+        description: Nonce for transactions.
+        tests:
+          - not_null
+
+      - name: public_key
+        description: The public key of an AccessKey which was used to sign the original transaction. In case of a deposit refund, the public key is empty (all bytes are 0).
+        tests:
+          - not_null
```
Comment (chuxinh) [inline models/core/actions_events_addkey.yml:33]: if this could be empty, maybe remove the test here

File: models/core/actions_events_addkey.yml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+      - name: permission
+        description: Permissions granted to the contract by the signer.
+        tests:
+          - not_null
+
+      - name: allowance
+        description: Amount of NEAR approved for use in unadjusted format (10^24). NULL if permission is FullAccess.
+        tests:
```
Comment (chuxinh) [inline models/core/actions_events_addkey.yml:42]: remove this line

File: models/core/actions_events_addkey.yml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+      - name: method_name
+        description: Name of the method(s) approved for use. NULL if permission is FullAccess.
+        tests:
```
Comment (chuxinh) [inline models/core/actions_events_addkey.yml:46]: remove this line

File: models/core/actions_events_addkey.yml [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+      - name: receiver_id
+        description: the account ID of the destination of this transaction. NULL if permission is FullAccess.
```
Comment (chuxinh) [inline models/core/actions_events_addkey.yml:49]:
```suggestion
        description: The account ID of the destination of this transaction. NULL if permission is FullAccess.
```

File: models/core/actions_events_addkey.yml [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+        tests:
```
Comment (chuxinh) [inline models/core/actions_events_addkey.yml:50]: remove this line

**Review by chuxinh — CHANGES_REQUESTED** [review]

File: models/core/actions_events_addkey.yml (hunk 4 above)
Comment (forgxyz) [reply models/core/actions_events_addkey.yml:16]: Yes, do I need to designate something here?

Comment (chuxinh) [reply models/core/actions_events_addkey.yml:16]: nope haha you've already done that in the config at top. just need to update the description here by removing the primary key part

Comment (forgxyz) [reply models/core/actions_events_addkey.yml:16]: Ahhhh thank you

**Comment by forgxyz** [comment]
I believe I addressed all of these, apologies if I missed one for re-review.

File: models/core/actions_events_addkey.yml [hunk 10]
```diff
@@ -3,7 +3,7 @@ version: 2
 models:
   - name: actions_events_addkey
     description: |-
-      This table extracts all action events from a transaction and stores the argument data under action_data.
+      This table extracts all AkkDey events from actions and stores the argument data under action_data.
```
Comment (chuxinh) [inline models/core/actions_events_addkey.yml:6]: typo? `Addkey`

**Review by chuxinh — APPROVED** [review]
LGTM after nits

File: models/core/actions_events_addkey.yml (hunk 10 above)
Comment (forgxyz) [reply models/core/actions_events_addkey.yml:6]: ty

**Pull request merged**